### PR TITLE
Merge some modules where it makes sense

### PR DESCRIPTION
**Pull request by Danappelxx**
- Middleware all in one module
- JSONMediatype -> JSON
- URLEncodedFormMediaType -> URLEncodedForm

### CODE REVIEW

**Comment by Danappelxx** [comment]
Maybe...
- `HTTP` should also contain `HTTPParser` and `HTTPSerializer`?
- Move `IP`, `TCP`, `HTTPServer`, `HTTPClient`, `File` under a new folder `IO`?

**Comment by paulofaria** [comment]
yes and yes.. but I think eventually the IO part will have to be extracted into a separate module altogether.

File: Sources/Quark/Quark+Venice/Quark+Venice.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+}
+
+extension FileResponder {
+    public init(path: String, headers: Headers = [:]) {
+        self.init(path: path, headers: headers, file: File.self)
+    }
+}
+
+// Warning: Due to a swift bug this has to be in the same file the protocol is declared
```
Comment (Danappelxx) [inline Sources/Quark/Quark+Venice/Quark+Venice.swift]: you should put a link to [this](https://github.com/QuarkX/Quark/blob/0de9bdf67e7b448e92dd1b675fe2277022c8c475/Sources/Quark/Quark/Resource/Resource.swift#L18) here, and vice versa.

File: Sources/Quark/Venice/File/File.swift [hunk 2]
```diff
@@ -301,92 +301,94 @@ extension File {
...
-        }
-
-        return currentPosition
-    }
-
-    func ends(with suffix: String) -> Bool {
-        return suffix == String(self.characters.suffix(suffix.characters.count))
-    }
-}
+// Warning: We're gonna need this when we split Venice from Quark in the future
```
Comment (Danappelxx) [inline Sources/Quark/Venice/File/File.swift]: Maybe a link to here as well?

File: Sources/Quark/Quark+Venice/Quark+Venice.swift (hunk 1 above)
Comment (paulofaria) [inline Sources/Quark/Quark+Venice/Quark+Venice.swift]: how?

**Pull request merged**